projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/upload/GitWrapper.java

```diff
@@ -108,13 +108,12 @@ private void pushChanges(
       String reason,
       PrePushOperation op)
       throws IOException, GitAPIException {
-    PushCommand push = git.push().setCredentialsProvider(credentialsProvider);
     op.apply();
     Status gitStatus = git.status().call();
     // if no changes are introduced, no point in try to push we skip
     if (gitStatus.hasUncommittedChanges()) {
       commitChanges(git, username, jobName, reason);
-      while (tryToPush(push)) {
+      while (tryToPush(git.push().setCredentialsProvider(credentialsProvider))) {
```

Comment:
@mivanov1988 
Within the latest version of jgit each git PushCommand object can only be called a single time. 
So each time we call `tryToPush` we need to make a new pushCommand object, as we can't call it more than once with the same instance. 

I notice that in your legacy code you create the push command before calling `op.apply()`. 
Is there a reason for this?

Reply:
Strange I did a git blame and it said it was you?!

Reply:
Most likely because the code predates VDK move to the open-source repo. And Miro moved the Control Service. And when moving we decided against maintaining the previous history. Marto (who is not working on VDK now) actually wrote this class. 

I suspect the reason is to avoid re-creating the same instance every time. But if it is needed this way, so be it.

Reply:
All the tests passed in main. I trust that we have good testing quality .